lib/portage/dbapi/bintree.py

```diff
@@ -1614,7 +1769,13 @@ def _max_build_id(self, cpv):
 		return max_build_id
 
 	def _allocate_filename(self, cpv):
-		return os.path.join(self.pkgdir, cpv + ".tbz2")
+		binpkg_format = self.settings.get("BINPKG_FORMAT", "xpak")
```

Comment:
`_allocate_filename` is similar to `getname` and ` _pkgindex_entry` we should not assume a constant binpkg_format here.